### PR TITLE
feat(tier4_perception_launch): make lanelet object filter optional

### PR DESCRIPTION
**Pull request by kaancolak**
Signed-off-by: Kaan Colak <kcolak@leodrive.ai>

## Description

When working without lanelet/vector map, we don't use any tier4_perception_launch configuration. use_vector_map is a parameter for the base launch files(autoware.launch.xml, logging_simulator.launch.xml) but this parameter doesn't affect "object_lanelet_filter".

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kaancolak** [comment]
@kenji-miyake Miyake-san, Could you review or assign an appropriate reviewer to this PR?

**Comment by kenji-miyake** [comment]
Sure. I think @yukkysaito and @miursh are the right developers to review this.

File: launch/tier4_perception_launch/launch/object_recognition/detection/camera_lidar_fusion_based_detection.launch.xml [hunk 1]
```diff
@@ -214,6 +215,8 @@
   </group>
 
   <group>
+    <let name="merger/output/objects" value="objects_before_filter" if="$(var use_vector_map)"/>
+    <let name="merger/output/objects" value="$(var output/objects)" unless="$(var use_vector_map)"/>
     <include file="$(find-pkg-share object_merger)/launch/object_association_merger.launch.xml">
       <arg name="input/object0" value="temporary_merged_objects"/>
       <arg name="input/object1" value="clustering/objects"/>
```
Comment (yukkysaito) [inline launch/tier4_perception_launch/launch/object_recognition/detection/camera_lidar_fusion_based_detection.launch.xml:222]: Is this work correctly? :pray:  
```
      <arg name="output/object" value="objects_before_filter"/>
```

I think the following code is correct
```
      <arg name="output/object" value="($var merger/output/objects)"/>
```

Comment (kaancolak) [reply launch/tier4_perception_launch/launch/object_recognition/detection/camera_lidar_fusion_based_detection.launch.xml:222]: Thanks,  @yukkysaito Saito-San. :pray: I tested with just lidar data. Missed this part.

Comment (yukkysaito) [reply launch/tier4_perception_launch/launch/object_recognition/detection/camera_lidar_fusion_based_detection.launch.xml:222]: Thank you for commit

**Review by yukkysaito — APPROVED** [review]
LGTM. (I don't confirm to run)
@miursh can you check the code?

**Comment by miursh** [comment]
@kaancolak In this PR, do you want to change not to use object_lanelet_filter or object_filter itself (including object_position_filter) ?
If you don't want to use object filter at final stage, arg name "use_vector_map" is kind of misleading, I think. "use_object_filter" is better.

**Comment by kaancolak** [comment]
@miursh Yes, I agree that it makes more sense to control the object filter with an extra parameter. I added and pushed the "use_object_filter" parameter.

**Review by miursh — APPROVED** [review]
LGTM

**Pull request merged**